src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/CachedExpressionCompiler.cs

```diff
@@ -11,12 +11,20 @@ namespace Microsoft.AspNetCore.Mvc.ViewFeatures
 {
     internal static class CachedExpressionCompiler
     {
+        private static readonly Expression NullExpression = Expression.Constant(value: null);
+        private static readonly MethodInfo ObjectReferenceEquals = typeof(object).GetMethod(
+            nameof(object.ReferenceEquals),
+            BindingFlags.Public | BindingFlags.Static,
+            binder: null,
+            new[] { typeof(object), typeof(object) },
+            modifiers: null);
+
         // This is the entry point to the cached expression compilation system. The system
         // will try to turn the expression into an actual delegate as quickly as possible,
         // relying on cache lookups and other techniques to save time if appropriate.
         // If the provided expression is particularly obscure and the system doesn't know
         // how to handle it, we'll just compile the expression as normal.
-        public static Func<TModel, TResult> Process<TModel, TResult>(
+        public static Func<TModel, object> Process<TModel, TResult>(
```

Comment:
`Process` returns two possible values - `null` if the expression or any of the `right` expressions evaluated to null, otherwise the value. `TResult` can be a value type e.g. `model => model.Customer.Id` and we cannot return the default value when the right expression evaluates to null. Converting it to an object is one solution, the alternative would be to turn this in to `Func<TModel, (bool, TResult)>`. Either solution requires rewriting the resulting delegate, and this is much easier to grock.